File: src/FsCheck.XUnit.QuickStart.CSharp/DollarExample.cs
using FsCheck.Xunit;

namespace FsCheck.QuickStart.XUnit.CSharp
{
    public class DollarExample
    {
        [Property]
        public bool SetAndGetShouldGiveSameResult(int amount)
        {
            var dollar = Dollar.Create(0);
            dollar.Add(amount);

            return dollar.Amount == amount;
        }

        [Property]
        public bool AddThenMultiplierSameAsCreate(int start, int times)
        {
            var dollar = Dollar.Create(0);
            dollar.Add(start);
            dollar.Multiplier(times);

            var dollar2 = Dollar.Create(start * times);

            return dollar.Amount == dollar2.Amount;
        }
        
        public class Dollar
        {
            private int _amount;

            public Dollar(int amount)
            {
                _amount = amount;
            } 
            
            public int Amount => _amount;
            
            public void Add(int add)
            {
                 _amount = _amount + add;
            }
            
            public void Multiplier(int multiplier)
            {
                _amount = _amount * multiplier;
            }
            
            public static Dollar Create(int amount)
            {
                return new Dollar(amount);
            }
        }
    }
}

File: src/FsCheck.XUnit.QuickStart.FSharp/DollarExample.fs
namespace FsCheck.XUnit.QuickStart.FSharp

module DollarModule =
    open FsCheck.Xunit

    type Dollar(amount : int) =
        let mutable privateAmount = amount;

        member this.Amount = privateAmount
        member this.Add add =
            privateAmount <- this.Amount + add
        member this.Times multiplier =
            privateAmount <- this.Amount * multiplier
        static member Create amount =
            Dollar amount

    [<Property>]
    let ``set then get should give same result`` value =
        let obj = Dollar.Create 0
        obj.Add value
        let newValue = obj.Amount
        value = newValue
        
    [<Property>]
    let ``add then multiplier same as create`` value times =
        let dollar = Dollar.Create 0
        dollar.Add value
        dollar.Times times
        
        let dollar2 = Dollar.Create(value*times);
        
        dollar.Amount = dollar2.Amount
